File: todoist-tests/src/test/java/il/co/topq/appium/CreateTaskTests.java
package il.co.topq.appium;

import java.io.File;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.List;
import java.util.concurrent.TimeUnit;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.DesiredCapabilities;

import io.appium.java_client.MobileElement;
import io.appium.java_client.android.AndroidDriver;
import io.appium.java_client.remote.MobileCapabilityType;

public class CreateTaskTests {

	private static final String EMAIL = "itai.agmon@top-q.co.il";
	private static final String PASSWORD = "secret";
	private AndroidDriver<MobileElement> driver;

	@Before
	public void setUp() throws MalformedURLException {
		File apkFile = new File("Todoist_com.todoist.apk");
		DesiredCapabilities capabilities = new DesiredCapabilities();
		capabilities.setCapability(MobileCapabilityType.VERSION, "1.4.0");
		capabilities.setCapability(MobileCapabilityType.PLATFORM_NAME, "Android");
		capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "6.0.1");
		capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android");
		capabilities.setCapability(MobileCapabilityType.APP, apkFile.getAbsolutePath());
		String activity = "com.todoist.activity.HomeActivity";
		capabilities.setCapability("appPackage", "com.todoist");
		capabilities.setCapability("appActivity", activity);
		capabilities.setCapability("appWaitActivity", "com.todoist.activity.WelcomeActivity");
		capabilities.setCapability("newCommandTimeout", "600");
		
		driver = new AndroidDriver<>(new URL("http://127.0.0.1:4723/wd/hub"), capabilities);
		driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
	}

	@Test
	public void testCreateNewTask() throws InterruptedException {
		// *** Welcome Activity ***
		driver.findElement(By.id("btn_welcome_log_in")).click();

		// *** Google Play Services Activity ***
		driver.findElement(By.id("button1")).click();

		// *** Login Activity ****
		driver.findElement(By.id("log_in_email")).sendKeys(EMAIL);
		driver.findElementById("log_in_password").sendKeys(PASSWORD);
		driver.findElementById("btn_log_in").click();

		// *** Google Play Services Activity ***
		driver.findElementById("button1").click();

		// Clicking on the Today menu item
		driver.findElement(By.xpath("//android.widget.TextView[@text='Today']")).click();
		
		// Creating a new task
		driver.findElementById("fab").click();
		String taskName = "Automated task" + System.currentTimeMillis();
		WebElement message = driver.findElementById("message");
		message.sendKeys(taskName + " #work");
		driver.findElementById("button1").click();

		// Going back
		driver.findElementById("action_mode_close_button").click();
		
		

		// Searching for the newly created item
		List<MobileElement> items = driver.findElements(By.xpath("//android.widget.RelativeLayout[@resource-id='com.todoist:id/item']/android.widget.TextView"));
		boolean found = false;
		for (WebElement item : items) {
			System.out.println("Item: " + item.getText());
			if (item.getText().equals(taskName)) {
				found = true;
				break;
			}
		}
		Assert.assertTrue("Item was not created", found);
	}

	@After
	public void tearDown() {
		if (driver != null) {
			driver.quit();
		}
	}

}

File: C#/TodoistTests/CreateTaskTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Appium;
using OpenQA.Selenium.Appium.Android;
using OpenQA.Selenium.Appium.Enums;
using OpenQA.Selenium.Remote;
using System;
using System.Collections.ObjectModel;
using System.IO;
using System.Threading;

namespace TodoistTests
{
    [TestClass]
    public class CreateTaskTests
    {
        private static String EMAIL = "itai.agmon@top-q.co.il";
	    private static String PASSWORD = "secret";

        AndroidDriver<AppiumWebElement> driver;

        [TestInitialize]
        public void Setup()
        {
            DesiredCapabilities capabilities = new DesiredCapabilities();
            capabilities.SetCapability(MobileCapabilityType.AppiumVersion, "1.4.0");
            capabilities.SetCapability(MobileCapabilityType.PlatformName, "Android");
            capabilities.SetCapability(MobileCapabilityType.PlatformVersion, "5.1.0");
            capabilities.SetCapability(MobileCapabilityType.DeviceName, "Android");
            string folder = Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory()));
            capabilities.SetCapability(MobileCapabilityType.App, folder + @"\Todoist_com.todoist.apk");
            String activity = "com.todoist.activity.HomeActivity";
            capabilities.SetCapability("appPackage", "com.todoist");
            capabilities.SetCapability("appActivity", activity);
            capabilities.SetCapability("appWaitActivity", "com.todoist.activity.WelcomeActivity");
            driver = new AndroidDriver<AppiumWebElement>(new Uri("http://127.0.0.1:4723/wd/hub"), capabilities);
        }

        [TestMethod]
        public void TestCreateSimpleTask()
        {
            driver.FindElement(By.Id("btn_welcome_log_in")).Click();
            // *** Google Play Services Activity ***
            driver.FindElement(By.Id("button1")).Click();

            // *** Login Activity ****
            driver.FindElement(By.Id("log_in_email")).SendKeys(EMAIL);
            driver.FindElementById("log_in_password").SendKeys(PASSWORD);
            driver.FindElementById("btn_log_in").Click();

            // *** Google Play Services Activity ***
            driver.FindElementById("button1").Click();
            
            // *** Goolge Play services in not supported ***
            //driver.FindElementById("button1").Click();

            // Clicking on the Today menu item
            driver.FindElement(By.XPath("//android.widget.TextView[@text='Today']")).Click();

            // Creating a new task
            driver.FindElementById("fab").Click();
            String taskName = "Automated task" + DateTime.Now.Ticks / TimeSpan.TicksPerMillisecond;
            AppiumWebElement message = driver.FindElementById("message");
            message.SendKeys(taskName + " #work");
            driver.FindElementById("button1").Click();

            // Going back
            driver.FindElementById("action_mode_close_button").Click();



            // Searching for the newly created item
            ReadOnlyCollection<AppiumWebElement> items = driver.FindElements(By.XPath("//android.widget.RelativeLayout[@resource-id='com.todoist:id/item']/android.widget.TextView"));
            Boolean found = false;
            foreach (AppiumWebElement item in items)
            {
                Console.Write("Item: " + item.Text);
                if (item.Text.Equals(taskName))
                {
                    found = true;
                    break;
                }
            }            
            Assert.IsTrue(found, "Item was not created");


            Thread.Sleep(1000);
        }

        [TestCleanup]
        public void Teardown()
        {
            driver.Quit();

        }
    }
}
